Sources/SwiftlyCore/HTTPClient.swift

```diff
@@ -5,23 +5,45 @@ import NIO
 import NIOFoundationCompat
 import NIOHTTP1
 
+public protocol ToolchainDownloader {
+    func downloadToolchain(
+        _ toolchain: ToolchainVersion,
+        url: String,
+        to destination: String,
+        reportProgress: @escaping (SwiftlyHTTPClient.DownloadProgress) -> Void
+    ) async throws
+}
+
 /// HTTPClient wrapper used for interfacing with various APIs and downloading things.
-public class HTTP {
+public class SwiftlyHTTPClient {
```

Comment:
Most of swiftly's test's runtime is devoted to downloading toolchains, and so in an effort to reduce that, I introduced the ability to mock toolchain downloads. This allows the update tests to run quickly while still verifying that updating works. We can consider expanding the use of mocking to other tests to reduce how long they run for, though I do think there is value in having the install command tests be non-mocked.

Reply:
Can we add some comment above `ToolchainDownloader` detailing why it is needed

Reply:
done